fmc/src/main.rs

```diff
@@ -109,6 +110,48 @@ fn report_error(code: u32) -> ! {
     }
 }
 
+/// Report fatal F/W error
+///
+/// # Arguments
+///
+/// * `val` - F/W error code.
+fn report_fw_error_fatal(val: u32) {
+    let mut soc_ifc = unsafe { SocIfcReg::new() };
+    soc_ifc.regs_mut().cptra_fw_error_fatal().write(|_| val);
+}
+
+#[allow(clippy::empty_loop)]
+fn handle_fatal_error(code: u32) -> ! {
```

Comment:
If we think this might be useful for RT FW, could put it in common/. Or could move it there once we get to that point in RT FW development.

Reply:
Sure. I can make this common on a follow up.